File: 2308-divide-array-into-equal-pairs/divide-array-into-equal-pairs.go
func divideArray(nums []int) bool {
    pairs := make(map[int]int);

    for _, num := range nums {
        pairs[num]++;
    }

    for _, pair := range pairs {
        
        if pair % 2 == 1 {
            return false;
        }
    }

    return true;
}

File: 2308-divide-array-into-equal-pairs/divide-array-into-equal-pairs.java
class Solution {
    public boolean divideArray(int[] nums) {
        int[] pairs = new int[501];

        for (int num : nums) {
            pairs[num]++;
        }

        for (int pair : pairs) {
            if ((pair & 1) != 0) return false;
        }

        return true;
    }
}